core/src/main/java/org/mule/runtime/core/context/notification/ConnectorMessageNotification.java

```diff
@@ -16,7 +17,7 @@
 /**
  * Used to notify that a message was received or sent through a Connector
  */
-public class ConnectorMessageNotification extends ServerNotification
+public class ConnectorMessageNotification extends ServerNotification implements BlockingServerEvent
```

Comment:
This was a refactor mistake than I'm fixing. You can check the 3.x branch to see the difference.